### PR TITLE
Mining tweaks.

### PR DESCRIPTION
**Pull request by Rail-Runner**
1. Allow mining downstairs/upstairs constructions to use electric jackhammer.
2. Make construction use jackhammer charges, not gasoline as a component.
3. Add more valid helmets for mining.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Do charges contained within a tool work in construction now? They didn't previously, and I'm not sure if that's changed.

**Comment by KA101** [comment]
Last I checked it drained gas charges from the gas jackhammer just fine, if that's what I had available?

File: data/json/construction.json [hunk 1]
```diff
@@ -2020,16 +2019,18 @@
...
         "tools" : [
-            [ "jackhammer", "jacqueshammer" ],
+            [ [ "jackhammer", 30 ], [ "jacqueshammer", 30 ], [ "elec_jackhammer", 30 ] ],
             [ "crowbar", "halligan"],
             [ "miner_hat", "hat_hard", "firehelmet", "helmet_ball",
-            "helmet_army", "helmet_lobster", "helmet_riot"]
+            "helmet_army", "helmet_lobster", "helmet_riot", "depowered_helmet", "helmet_bone",
+            "helmet_chitin", "helmet_hsurvivor", "helmet_survivor", "helmet_xlsurvivor", "miner_hat_on",
+            "power_armor_helmet_basic", "power_armor_helmet_heavy", "power_armor_helmet_light", "tac_fullhelmet",
+            "helmet_barbute", "helmet_plate", "helmet_kabuto"]
```
Comment (KA101) [inline data/json/construction.json]: I'm gonna suggest that the helmets should be open-face or powered-on.  Mining is tough work and it helps if you can breathe easily.

Comment (Darkyhard) [inline data/json/construction.json]: Well, maybe survivor stops mining for some time, takes their helmet off, rests and then keeps mining? It's pretty much impossible to do such a hard work for 8 hours without any rest anyway.

Comment (KA101) [inline data/json/construction.json]: Small-scale rests are involved in changing from jackhammer to shovel, etc.  Basically, I'm content with requiring miners use some reasonable facsimile of mining gear.  This ain't Terraria.

Comment (Darkyhard) [inline data/json/construction.json]: Okay, done.

Comment (KA101) [inline data/json/construction.json]: Thanks.

**Comment by KA101** [comment]
Out of time error, sorry.

**Comment by Darkyhard** [comment]
So, something is wrong with this? There is already a metal bars construction in game which uses welder charges, so I can't think of why these changes don't work.

**Comment by KA101** [comment]
Out of time means that I was tied up dealing with forumite crap and didn't have a chance to properly test this.

**Pull request merged**